core/controllers/profile_test.py

```diff
@@ -282,3 +283,27 @@ def test_profile_data_is_independent_of_currently_logged_in_user(self):
         response = self.get_json(
             '/profilehandler/data/%s' % self.EDITOR_USERNAME)
         self.assertEqual(response['user_bio'], 'My new editor bio')
+
+class FirstContributionDateTests(test_utils.GenericTestBase):
+
+    USERNAME = 'abc123'
+    EMAIL = 'abc123@gmail.com'
+
+    def test_contribution_datetime(self):
+        """Test the contribution date shows up correctly as nonexist."""
+        self.signup(self.EMAIL, self.USERNAME)
+        self.login(self.EMAIL)
+        user_services.get_or_create_user(self.USERNAME, self.EMAIL)
+        self.user_id = self.get_user_id_from_email(self.EMAIL)
+
+        response = self.testapp.get('/profile/%s' % self.USERNAME)
+        self.assertEqual(response.status_int, 200)
+        response.mustcontain(self.USERNAME)
+
+        user_services.set_contribution_date(self.user_id, datetime.utcnow())
+
+        """Test the contribution date correctly changes to set date time."""
+        response = self.testapp.get('/profile/%s' % self.USERNAME)
+        print response
+        self.assertEqual(response.status_int, 200)
+        response.mustcontain('Contributing since')
```

Comment:
This doesn't check that the correct datetime shows. It only checks that the string "Contributing since" is in the HTML page returned by the GET handler corresponding to /profile; this is going to be true in any case.

Instead, here, you should check the output of get_json for /profilehandler and verify the keys and values it contains.